lib/spack/spack/solver/concretize.lp

```diff
@@ -1010,23 +997,24 @@ variant_is_propagated(PackageNode, Variant) :-
 % 1. The same flag type is not set on this node
 % 2. This node has the same compiler as the propagation source
 
-propagated_flag(node(PackageID, Package), node_flag(FlagType, Flag), SourceNode) :-
-  propagate(node(PackageID, Package), node_flag(FlagType, Flag), _),
-  not attr("node_flag_set", node(PackageID, Package), FlagType, _),
+propagated_flag(node(PackageID, Package), node_flag(FlagType, Flag, FlagGroup, Source), SourceNode) :-
+  propagate(node(PackageID, Package), node_flag(FlagType, Flag, FlagGroup, Source), _),
+  % Disable to pass propagate_and_spec_dep, propagate_and_pkg_dep
+  % and propagate_and_require
+  % not attr("node_flag_set", node(PackageID, Package), FlagType, _, _, _),
```

Comment:
(the reincarnation of https://github.com/spack/spack/pull/41049/files#r1591611535)

Reply:
Resolved in https://github.com/spack/spack/pull/41049#discussion_r1732076592